docs/source/python/feather.rst

```diff
@@ -0,0 +1,107 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow
+
+.. _feather:
+
+Feather File Format
+===================
+
+Feather is a portable file format for storing Arrow tables or data frames (from
+languages like Python or R) that utilizes the :ref:`Arrow IPC format <ipc>`
+internally. Feather was created early in the Arrow project as a proof of
+concept for fast, language-agnostic data frame storage for Python (pandas) and
+R. There are two file format versions for Feather:
+
+* Version 2, the default version, which is exactly represented as the Arrow IPC
+  file format on disk. Version 2 files support storing all Arrow data types as
+  well as compression with LZ4 or ZSTD. Version 2 was first made available in
+  Apache Arrow 0.17.0.
+* Version 1, a legacy version available starting in 2016, replaced by
+  Version 2. Version 1 files are distinct from Arrow IPC files and lack many
+  features, such as the ability to store all Arrow data types. Version 1 files
+  also lack compression support.
```

Comment:
Maybe mention explicitly that Version 1 files can still be read for the foreseeable future? 
(which I suppose will be the first question many people that already have existing feather files will ask themselves)